### DISCUSSION TITLE
incubator-session, incubator-cli, etc.: When for v5?

### DISCUSSION THREAD

**Discussion by not-art** [phalcon/cphalcon#15873] [Q&A] [answered]
Hi phalcon team, both official & community contributors!

My company uses phalcon as a framework in roughly 1/3rd of our application stack, and we're looking to upgrade given that we're on v3 w/ php73, which is no-longer receiving security updates.

As-such, our options are v4, v5, or a new framework entirely. We'd like to go with v5, to avoid needing to make two non-trivial phalcon upgrades within the same year; if we go v4, and thus php74, we're forced to ALSO go v5 before November of this year, as php74's already only receiving security updates, which stops in November 2022.

The issue, however, is we have made decent use of phalcon incubator, which as of v4, is now broken out in it's own specific repos, eg: incubator-session.

These haven't seen any development work since October, however, and none of them seem to have branches where work for v5 support is taking place / they only support up-to v4. Which, unfortunately, means we'd be looking at needing to go v3->v4 soon, and v4->v5 sometime before fall of this year.

**What is the plan with these incubator-* repos / how soon until these incubator-* repos are updated to support phalcon v5?**

Thanks in advance & for all the work put into the framework over the years,
-Art

**Comment by not-art** [comment, marked as answer]
Jeckerson answered my question on the discord- if anyone is curious about the same question:

> incubator, devtools and migrations will start to be upgraded as soon as v5 releases

So if you're running a v3 project dependent on any of the incubator projects, you're going to have to wait until v5 leaves beta (any week now, hopefully!).

**Reply by niden** [reply]
The main focus for v5 for now is 
- one feature remaining (paginator)
- php 8.1 support (some of the interfaces are a hell to adjust)
- documentation, documentation documentation

Jeckerson is handing the PHP 8.1 and I am doing the rest (although I have been slacking a bit lately).

Hopefully in the next few weeks we can iron out all the documents and especially the upgrade guide and then start working on the satellite applications such as incubator repos, invo, vokuro etc.
